From f5b4b0abb8f98ec6a746f8d2f302f3534cd51bd5 Mon Sep 17 00:00:00 2001
From: heyzec <61238538+heyzec@users.noreply.github.com>
Date: Wed, 2 Oct 2024 18:43:01 +0800
Subject: [PATCH 1/4] add simple message passing for editor sync

---
 src/index.tsx | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/src/index.tsx b/src/index.tsx
index bb990d08..9a612caa 100644
--- a/src/index.tsx
+++ b/src/index.tsx
@@ -15,6 +15,16 @@ import { store } from 'src/pages/createStore';
 import ApplicationWrapper from './commons/application/ApplicationWrapper';
 import { createInBrowserFileSystem } from './pages/fileSystem/createInBrowserFileSystem';
 
+window.addEventListener('message', (event) => {
+  const elements = document.getElementsByClassName('react-ace');
+  if (elements.length === 0) {
+    return;
+  }
+  // @ts-expect-error: ace is not available at compile time
+  const editor = ace.edit(elements[0]);
+  editor.setValue(event.data);
+});
+
 if (Constants.sentryDsn) {
   Sentry.init({
     dsn: Constants.sentryDsn,
-- 
2.47.0

